decrypt.go

```diff
@@ -280,8 +279,8 @@ func (ds *decryptStream) processEncryptionBlock(bl *encryptionBlock) ([]byte, er
 
 	// Check the authenticator.
 	hashToAuthenticate := computePayloadHash(ds.headerHash, nonce, ciphertext)
-	ourAuthenticator := hmacSHA512256(ds.macKey, hashToAuthenticate)
```

Comment:
Ugh sorry, this function is horribly named. (I patterned it after https://nacl.cr.yp.to/auth.html, which I regret now.)